CRM/Export/BAO/Export.php

```diff
@@ -760,12 +757,8 @@ public static function exportComponents(
 
     // for CRM-3157 purposes
     $i18n = CRM_Core_I18n::singleton();
-    $outputColumns = array();
-    //@todo - it would be clearer to start defining output columns earlier in this function rather than stick with return properties until this point
-    // as the array is not actually 'returnProperties' after the sql query is formed - making the alterations to it confusing
-    foreach ($returnProperties as $key => $value) {
-      $outputColumns[$key] = $value;
-    }
+    list($outputColumns, $headerRows, $sqlColumns, $metadata) = self::getExportStructureArrays($returnProperties, $query, $phoneTypes, $imProviders, $contactRelationshipTypes, $relationQuery, $selectedPaymentFields);
```

Comment:
A disadvantage of this approach is you miss the chance to take advantage of $outputColumns being looped through later & parasiting onto that loop. The advantage however, is that if that loop runs 100,000 times it doesn't have to decide not to extract the headers 999,999 times...